libs/isograph-react/src/core/cache.ts

```diff
@@ -149,15 +153,20 @@ export function normalizeData(
       variables,
     );
   }
+
+  environment.store[root.concreteType] ??= {};
+  environment.store[root.concreteType]![root.id] ??= {};
```

Comment:
I had a change of heart: Since we have so many multi-level maps, I guess the `??=` operator is quite useful. So let's just embrace it.

Reply:
I like these shorthands, because when the value on the left is long, then it takes some time to understand that value on the right is the same
```tsx
  cachedComponentsById[
    fragmentReference.root.__typename + ':' + fragmentReference.root.__link
  ] =
    cachedComponentsById[
      fragmentReference.root.__typename + ':' + fragmentReference.root.__link
    ] ?? {};
```